src/c/coap-server.c

```diff
@@ -0,0 +1,465 @@
+/* CoAP server for device-coap-c
+ *
+ * Copyright (C) 2018 Olaf Bergmann <bergmann@tzi.org>
+ * Copyright (c) 2020 Ken Bannister
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <float.h>
+#include <signal.h>
+#include <string.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <netdb.h>
+
+#include <coap2/coap.h>
+#include "edgex/devices.h"
+#include "device-coap.h"
+
+/* Maximum length of a string containing numeric values. */
+#define INT32_STR_MAXLEN 10
+#define FLOAT64_STR_MAXLEN (DBL_MAX_10_EXP + 2)
```

Comment:
Increase INT32_STR_MAXLEN to 11 to account for a negative integer.

DBL_MAX_10_EXP is 308. It was a mistake to use this value as the basis for FLOAT64_STR_MAXLEN. Changing to 24 to expect use of exponential notation for really large/small values. So an example for a max length string is: -2.1234567890123456E-190. In the future we should accept a more compact encoding.

Fixed in dde2b30